faucet/src/faucet.rs

```diff
@@ -230,17 +292,11 @@ pub fn request_airdrop_transaction(
             "request_airdrop_transaction: buffer length read_exact error: {:?}",
             err
         );
-        Error::new(ErrorKind::Other, "Airdrop failed")
+        err
     })?;
     let transaction_length = LittleEndian::read_u16(&buffer) as usize;
     if transaction_length > PACKET_DATA_SIZE || transaction_length == 0 {
-        return Err(Error::new(
-            ErrorKind::Other,
-            format!(
-                "request_airdrop_transaction: invalid transaction_length from faucet: {}",
-                transaction_length
-            ),
-        ));
+        return Err(FaucetError::InvalidTransactionLength(transaction_length));
```

Comment:
It might be nice to differentiate between `transaction_length > PACKET_DATA_SIZE` (`FaucetError::TransactionDataTooLarge`?)  and `transaction_length == 0` (`FaucetError::NoDataReceived`?)